sample-app/src/components/VisualAutocomplete/VisualSearchBar.tsx

```diff
@@ -0,0 +1,157 @@
+  const [visualAutocompleteState, executeVisualEntitiesQuery] = useVisualEntities(headlessId, debounceTime);
+  const { verticalResultsArray, isLoading: autocompleteLoading } = visualAutocompleteState;
+
+  const [ executeQuery, autocompletePromiseRef ] = useNearMeSearch(answersActions)
+
+  const [autocompleteResponse, executeAutocomplete] = useSynchronizedRequest(async () => {
+    return answersActions.executeUniversalAutocomplete();
+  });
+
+  const options = autocompleteResponse?.results.map(result => {
+    return {
+      value: result.value,
+      display: renderWithHighlighting(result)
+    }
+  }) ?? [];
+
+  const screenReaderText = processTranslation({
+    phrase: `${options.length} autocomplete option found.`,
+    pluralForm: `${options.length} autocomplete options found.`,
+    count: options.length
+  });
+
+  const restrictedVerticals = useMemo(() => calculateRestrictedVerticals(children), [children]);
+  const universalLimit = useMemo(() => calculateUniversalLimits(children), [children]);
+  // TODO: add restrictedVerticals to answers-headless and use it here in next PR
+
+  function renderDropdownSection() {
+    return <DropdownSection
+      options={options}
+      optionIdPrefix='VisualAutocompleteOption_0'
+      onFocusChange={value => {
+        answersActions.setQuery(value);
+        executeVisualEntitiesQuery(value, universalLimit);
+      }}
+      onSelectOption={optionValue => {
+        autocompletePromiseRef.current = undefined;
+        answersActions.setQuery(optionValue);
+        executeQuery();
+      }}
+      cssClasses={{
+        sectionContainer: 'Autocomplete__dropdownSection',
+        sectionLabel: 'Autocomplete__sectionLabel',
+        optionsContainer: 'Autocomplete_sectionOptions',
+        option: 'Autocomplete__option',
+        focusedOption: 'Autocomplete__option--focused'
+      }}
+    />
+  }
+
+  return (
+    <div className='SearchBar'>
+      <InputDropdown
+        inputValue={query}
+        placeholder={placeholder}
+        screenReaderInstructions={SCREENREADER_INSTRUCTIONS}
+        screenReaderInstructionsId={screenReaderInstructionsId}
+        screenReaderText={screenReaderText}
+        onSubmit={executeQuery}
+        onInputChange={value => {
+          answersActions.setQuery(value);
+          executeVisualEntitiesQuery(value, universalLimit);
+        }}
+        onInputFocus={() => {
+          autocompletePromiseRef.current = executeAutocomplete();
+        }}
+        renderButtons={() => 
+          <SearchButton
+            className='SearchBar__submitButton'
+            handleClick={executeQuery}
+            isLoading={isLoading}
+          />
+        }
+        cssClasses={{
+          dropdownContainer: 'Autocomplete',
+          inputElement: 'SearchBar__input',
+          inputContainer: 'SearchBar__inputContainer'
+        }}
+      >
+        {options.length > 0 && renderDropdownSection()}
```

Comment:
One thing that feels odd to me is the concept of the `VisualAutocompleteSection`. When I see something like that, I would assume we could use it to represent all pieces of the drop-down experience, including query suggestions. But, query suggestions seems to be its own special flower. Will recent searches have to be a special flower as well, or can we use a Section for that?

Reply:
IMO, if `VisualAutocompleteSection` can only be used for entity previews, we should just invoke the `VisualAutocompleteEntities` Component directly in the `props.children` of the `VisualSearchBar`.